packages/simcore-sdk/src/simcore_sdk/node_ports_common/file_io_utils.py

```diff
@@ -185,16 +185,25 @@ async def download_link_to_file(
         reraise=True,
         wait=wait_exponential(min=1, max=10),
         stop=stop_after_attempt(num_retries),
-        retry=retry_if_exception_type(ClientConnectionError),
+        retry=retry_if_exception_type(httpx.TransportError),
         before_sleep=before_sleep_log(_logger, logging.WARNING, exc_info=True),
         after=after_log(_logger, log_level=logging.ERROR),
     ):
         with attempt:
             async with AsyncExitStack() as stack:
-                response = await stack.enter_async_context(session.get(url))
-                if response.status == status.HTTP_404_NOT_FOUND:
+                client = await stack.enter_async_context(
+                    httpx.AsyncClient(
```

Comment:
creating a new client everytime is costly and wasting resources.
here you are changing the behavior. I don't want to stop this now, but please take care of that in a next PR, thank you.
Create an issue and let's fix this calmly.